creator-node/src/snapbackSM.js

```diff
@@ -47,63 +58,308 @@ class SnapbackSM {
     )
   }
 
+  // Helper function to retrieve all config based
+  async getSPInfo () {
+    const spID = config.get('spID')
+    const endpoint = this.endpoint
+    const delegateOwnerWallet = config.get('delegateOwnerWallet')
+    const delegatePrivateKey = config.get('delegatePrivateKey')
+    return {
+      spID,
+      endpoint,
+      delegateOwnerWallet,
+      delegatePrivateKey
+    }
+  }
+
+  // Query eth-contracts chain for endpoint to ID info
+  async recoverSpID () {
+    if (config.get('spID') !== 0) {
+      this.log(`Known spID=${config.get('spID')}`)
+      return
+    }
+
+    const recoveredSpID = await this.audiusLibs.ethContracts.ServiceProviderFactoryClient.getServiceProviderIdFromEndpoint(
+      this.endpoint
+    )
+    this.log(`Recovered ${recoveredSpID} for ${this.endpoint}`)
+    config.set('spID', recoveredSpID)
+  }
+
+  // Retrieve users with this node as primary
+  async getNodePrimaryUsers () {
+    const currentlySelectedDiscProv = this.audiusLibs.discoveryProvider.discoveryProviderEndpoint
+    let requestParams = {
+      method: 'get',
+      baseURL: currentlySelectedDiscProv,
+      url: `users/creator_node`,
+      params: {
+        creator_node_endpoint: this.endpoint
+      }
+    }
+    let resp = await axios(requestParams)
```

Comment:
any danger of this response exceeding max size?

Reply:
have not observed against production data after 24hrs, still tbd though